### DISCUSSION TITLE
Best way to display a difference or duration with native Date

### DISCUSSION THREAD

**Discussion by erwanlpfr** [lingui/js-lingui#1768] [General]
Hello, 

I just found your lib and I find it really easy and flexible. I converted my whole app with it!
I am wondering how to display a difference between two dates in the most native way.

Actually, I use a pretty heavy syntax to do so.

```typescript
  useEffect(() => {
    const interval = setInterval(() => {
      const diff = (Date.now() - props.startTime) / 1000;
      const hours = Math.floor((diff / (60 * 60)) % 24);
      const minutes = Math.floor((diff / 60) % 60);
      const seconds = Math.floor(diff % 60);

      setCounterDate([
        (hours < 10 ? '0' : '') + hours,
        (minutes < 10 ? '0' : '') + minutes,
        (seconds < 10 ? '0' : '') + seconds,
      ]);
    }, 1000);

    return () => clearInterval(interval);
  }, [props.startTime]);
```

I come to you because I feel there is a better solution that I don't see.
Is there any "native" solution to display it ?

It would be like that :

```typescript
          {i18n.date(duration, {
            hour: '2-digit',
            minute: '2-digit',
            second: '2-digit',
          })}
```

Thank you !

**Comment by timofei-iatsenko** [comment]
I don't see any better solutin without using helper libraries. 

BTW there are [RelativeTimeFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/RelativeTimeFormat) which is widely adopted, and upcoming experimental [DurationFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DurationFormat)
